addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -533,22 +546,133 @@ var ImageWidget = MediaWidget.extend({
             }
 
             function _processFile(attachment, error) {
-                var $button = self.$('.o_upload_image_button');
                 if (!error) {
-                    $button.addClass('btn-success');
                     self._toggleImage(attachment, true);
                 } else {
-                    $button.addClass('btn-danger');
                     self.$el.addClass('o_has_error').find('.form-control, .custom-select').addClass('is-invalid');
                     self.$el.find('.form-text').text(error);
                 }
-
-                if (!self.multiImages) {
-                    self.trigger_up('save_request');
-                }
             }
         });
         this.$el.submit();
+        // Emptying file inputs
+        this.$('#imagefile').val('');
+        this.$('#documentfile').val('');
+    },
+    /**
+     * @private
+     */
+    _loadMoreImages: function () {
+        this.IMAGES_ROWS += 2;
+        this.IMAGES_PER_PAGE = this.IMAGES_ROWS * this.IMAGES_PER_ROW;
+        this._renderImages();
+        this._displayLoadMoreBtnImages();
+    },
+    /**
+     * @private
+     */
+    _checkImageUrl: function () {
+        var self = this;
+        var formats = ['.gif', '.jpe', '.jpg', '.png'];
+        var url = this.$('#imageurl').val();
+        var $warning = this.$('.warning-url-image');
+        var $success = this.$('.success-url-image');
+        var $btn = this.$('.o_upload_image_url_button');
+        var regex = /^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/;
+        var success = url.match(regex) !== null;
+
+        if (success) {
+            _.each(formats, function (format) {
+                var endsWith = url.endsWith(format);
+                if (endsWith) {
+                    $warning.toggleClass('d-none', success);
+                    $success.toggleClass('d-none', !success);
+                    self._nameButton('image', 'images');
+                    $btn.toggleClass('disabled', !success);
+                }
+            });
+        } else {
+            $warning.toggleClass('d-none', success);
+            $success.toggleClass('d-none', !success);
+            this._nameButton('document', 'images');
```

Comment:
I think you did not understand the previous comment. Here you are using toggleClass in a context where the `success` variable is always `false`... so you could use `addClass` and `removeClass` directly. But I still think this should not be done this way. The whole function should **end** with those two `toggleClass` and using a boolean variable which is the result of all what has been checked above.

Also, this whole method is very close to `_checkDocumentURL`, I think what you want is a generic `_checkURL` handling both cases and using generic css classes to target the DOM elements.